src/Illuminate/Database/Eloquent/Concerns/HasAttributes.php

```diff
@@ -1180,11 +1182,16 @@ protected function getEnumCaseFromValue($enumClass, $value)
     /**
      * Get the storable value from the given enum.
      *
+     * @param  string  $expectedEnum
      * @param  \UnitEnum|\BackedEnum  $value
      * @return string|int
      */
-    protected function getStorableEnumValue($value)
+    protected function getStorableEnumValue($expectedEnum, $value)
```

Comment:
This is a breaking change unfortunately. We can't change a method signature in a patch release.

Reply:
Can we think of an alternative way of accomplishing this that wont be a breaking change?

Reply:
The only alternative I can see would be to add the `instanceof` check before calling `getStorableEnumValue()` (i.e. in `addCastAttributesToArray()` and `setEnumCastableAttribute()`).

The downside of this is duplicate code and if `getStorableEnumValue()` were to be added anywhere else in the future we'd need to remember to add the `instanceof` check there as well.

This doesn't seem like a good idea to me.

Reply:
If we can't do it differently then yeah master might be the only option